homework/advanced-calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,190 @@
+#include "advancedCalculator.hpp"
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <string>
+
+const std::string operationSigns = "+*/-%!^$";
+const std::string unalowedOperationSigns = "\"#&'():;<=>?@[\\]_`";
+const std::string digits = "1234567890";
```

Comment:
constexpr const char str[]. Since c++17 you can create constexpr std::string_view and since c++20 constexpr std::string :)